library/common/extensions/filters/http/platform_bridge/filter.cc

```diff
@@ -325,7 +369,20 @@ PlatformBridgeFilter::encodeTrailers(Http::ResponseTrailerMap& trailers) {
   return status;
 }
 
+void PlatformBridgeFilter::resumeDecoding() {
+  auto weak_self = weak_from_this();
+  dispatcher_.post([weak_self]() -> void {
+    if (auto self = weak_self.lock()) {
+      self->onResumeDecoding();
+    }
+  });
+}
+
 void PlatformBridgeFilter::onResumeDecoding() {
+  if (iteration_state_ == IterationState::Ongoing) {
+    return;
```

Comment:
In other parts of the filter we have assertions about invalid combination of actions vs filter state. I want to hear your thoughts about why here we have a no-op.

Reply:
In my view, It's expected that this could happen, and it shouldn't be treated as an error. Say you have a filter that halts iteration for the next data chunk only for up to a 30s timeout and then resumes regardless. You'd need to resume asynchronously after the 30 seconds are up, but you might get the next data chunk right around then too. The ordering of the dispatch of those two events is unpredictable. You would know if the async call won during your on-data call, because you'd have received an on-resume call (on the same thread - the stream/filter thread). But if the inverse was true, and you wanted to proceed, you'd have no way to abort the already-dispatched asynchronous call (at least not without making the implementation a whole lot more complicated). This allows you to use async resume in conjunction with sync resume without a whole lot of extra bookkeeping in the filter implementation and said complexity in the implementation.